smithy-build/src/main/java/software/amazon/smithy/build/SmithyBuildImpl.java

```diff
@@ -283,13 +283,24 @@ private ProjectionResult applyProjection(
                 .model(projectedModel)
                 .events(modelResult.getValidationEvents());
 
+        // Keep track of the first error created by plugins to fail the build after all plugins have run
+        Throwable firstPluginError = null;
```

Comment:
Will this only keep track of a single plugin error? This might be confusing if multiple plugins fail, but only the error for the first (or last?) ends up being thrown at the end.

Reply:
This only keeps track of the first plugin error. My initial thinking was to keep track of all plugin errors, but errors are handled per-projection (in [`SmithyBuild`](https://github.com/awslabs/smithy/blob/513f05fc34b236d40bcf0c1be849f5e25b90eaf4/smithy-build/src/main/java/software/amazon/smithy/build/SmithyBuild.java#L117) and [`BuildCommand`](https://github.com/awslabs/smithy/blob/513f05fc34b236d40bcf0c1be849f5e25b90eaf4/smithy-cli/src/main/java/software/amazon/smithy/cli/commands/BuildCommand.java#L206)). In order to not change these, we would need to aggregate all plugin errors into one error to throw as the projection error, but I wanted to defer making a decision on what exactly this would look like. Every plugin error is logged however, see [here](https://github.com/awslabs/smithy/pull/1416/commits/1fc7385fd15da01bd98b1463fbdd97e8e680f3ed#diff-e2603e6b410f984d3b14021df3820bac9de49da2db254ff5834522279283fba0R294), but I think it would be nice to throw one big exception with all plugin errors

Reply:
That sounds like a bigger change that could be done later, but this still seems like a good improvement for now.